server/src/main/java/org/cloudfoundry/identity/uaa/oauth/provider/config/xml/AuthorizationServerBeanDefinitionParser.java

```diff
@@ -0,0 +1,342 @@
+				.genericBeanDefinition(UaaTokenEndpoint.class);
+		tokenEndpointBean.addPropertyReference("clientDetailsService", clientDetailsRef);
+		tokenEndpointBean.addPropertyReference("tokenGranter", tokenGranterRef);
+		authorizationEndpointBean.addPropertyReference(O_AUTH_2_REQUEST_VALIDATOR,
+				oAuth2RequestValidatorRef);
+		parserContext.getRegistry().registerBeanDefinition("uaaTokenEndpoint",
+				tokenEndpointBean.getBeanDefinition());
+		if (StringUtils.hasText(oAuth2RequestFactoryRef)) {
+			tokenEndpointBean.addPropertyReference("oAuth2RequestFactory",
+					oAuth2RequestFactoryRef);
+		}
+		if (StringUtils.hasText(oAuth2RequestValidatorRef)) {
+			tokenEndpointBean.addPropertyReference(O_AUTH_2_REQUEST_VALIDATOR,
+					oAuth2RequestValidatorRef);
+		}
+
+		// Register a handler mapping that can detect the auth server endpoints
+		BeanDefinitionBuilder handlerMappingBean = BeanDefinitionBuilder
+				.rootBeanDefinition(FrameworkEndpointHandlerMapping.class);
+		ManagedMap<String, TypedStringValue> mappings = new ManagedMap<>();
+		if (StringUtils.hasText(tokenEndpointUrl)
+				|| StringUtils.hasText(authorizationEndpointUrl)) {
+			if (StringUtils.hasText(tokenEndpointUrl)) {
+				mappings.put("/oauth/token",
+						new TypedStringValue(tokenEndpointUrl, String.class));
+			}
+			if (StringUtils.hasText(authorizationEndpointUrl)) {
+				mappings.put("/oauth/authorize",
+						new TypedStringValue(authorizationEndpointUrl, String.class));
+			}
+			if (StringUtils.hasText(approvalPage)) {
+				mappings.put("/oauth/confirm_access",
+						new TypedStringValue(approvalPage, String.class));
+			}
+		}
+		if (StringUtils.hasText(enableCheckToken) && enableCheckToken.equals("true")) {
+			// configure the check token endpoint
+			BeanDefinitionBuilder checkTokenEndpointBean = BeanDefinitionBuilder
+					.rootBeanDefinition(CheckTokenEndpoint.class);
+			checkTokenEndpointBean.addConstructorArgReference(tokenServicesRef);
+			parserContext.getRegistry().registerBeanDefinition("oauth2CheckTokenEndpoint",
+					checkTokenEndpointBean.getBeanDefinition());
+
+			if (StringUtils.hasText(checkTokenUrl)) {
+				mappings.put("/oauth/check_token",
+						new TypedStringValue(checkTokenUrl, String.class));
+			}
+		}
+		if (!mappings.isEmpty()) {
+			handlerMappingBean.addPropertyValue("mappings", mappings);
+		}
+
+		if (StringUtils.hasText(approvalParameter) && registerAuthorizationEndpoint) {
+			if (!StringUtils.hasText(userApprovalHandlerRef)) {
+				BeanDefinitionBuilder userApprovalHandler = BeanDefinitionBuilder
+						.rootBeanDefinition(DefaultUserApprovalHandler.class);
+				userApprovalHandler.addPropertyValue("approvalParameter",
+						new TypedStringValue(approvalParameter, String.class));
+				authorizationEndpointBean.addPropertyValue("userApprovalHandler",
+						userApprovalHandler.getBeanDefinition());
+			}
+			handlerMappingBean.addPropertyValue("approvalParameter", approvalParameter);
+		}
+
+		// We aren't defining a filter...
```

Comment:
@strehle Can you elaborate on this comment? I assume it is from the original

Reply:
it is original yes therefore hard to say what it means, we can remove it, because after such a long time I dont know if it makes sense to have it as remark